src/Native/Signal.js

```diff
@@ -118,11 +118,19 @@ Elm.Native.Signal.make = function(localRuntime) {
     input.kids.push(this);
   }
 
-  function timestamp(a) {
-    function update() { return Utils.Tuple2(localRuntime.timer.now(), a.value); }
-    return new LiftN(update, [a]);
+  function Timestamp(input) {
+    this.id = Utils.guid();
+    this.value = Utils.Tuple2(localRuntime.timer.programStart, input.value);
+    this.kids = [];
+    this.recv = function(timestep, changed, parentID) {
+      if (changed) { this.value = Utils.Tuple2(timestep, input.value); }
```

Comment:
Sure, I can do these style changes. I had refrained from doing any of that, both to make the syntactic diff as small as possible and since you somewhere expressed that contributors should as much as possible preserve syntactic style they encounter in the files they edit.